src/core/time.d

```diff
@@ -2108,21 +2108,19 @@ struct MonoTimeImpl(ClockType clockType)
 
         version (Windows)
         {
-            long ticks;
-            if (QueryPerformanceCounter(&ticks) == 0)
-            {
-                // This probably cannot happen on Windows 95 or later
-                import core.internal.abort : abort;
-                abort("Call to QueryPerformanceCounter failed.");
-            }
+            long ticks = void;
+            QueryPerformanceCounter(&ticks);
             return MonoTimeImpl(ticks);
         }
         else version (Darwin)
             return MonoTimeImpl(mach_absolute_time());
         else version (Posix)
         {
-            timespec ts;
-            if (clock_gettime(clockArg, &ts) != 0)
+            timespec ts = void;
+            static if (ts.tv_sec.max >= long.max)
+                // clockArg is supported and overflow won't happen before 2.82e11 A.D.
+                clock_gettime(clockArg, &ts);
+            else if (clock_gettime(clockArg, &ts) != 0)
```

Comment:
Please adjust both of the sections so that there's only one call to `clock_gettime` e.g. something like

            timespec ts = void;
            immutable result = clock_gettime(clockArg, &ts);
            // clockArg is supported, and if tv_sec is long or larger, overflow
            // won't happen before 2.82e11 A.D.
            static if (ts.tv_sec.max < long.max)
            {
                if (result != 0)
                {
                    import core.internal.abort : abort;
                    abort("Call to clock_gettime failed.");
                }
            }

Repeating the calls like that just seems needlessly risky with regards to long term maintenance when the only difference that's really needed is whether the return value is actually read. There shouldn't be any difference in the resulting binary.

Reply:
Done.